src/services/counters/CounterService.ts

```diff
@@ -0,0 +1,176 @@
+import {App} from "../../ManagementConsole";
+import {DmrService} from "../dmr/DmrService";
+import {ICounter} from "./ICounter";
+import {IDmrRequest} from "../dmr/IDmrRequest";
+import {SocketBindingService} from "../socket-binding/SocketBindingService";
+import IQService = angular.IQService;
+import {LaunchTypeService} from "../launchtype/LaunchTypeService";
+import {isNotNullOrUndefined, traverse, deepValue} from "../../common/utils/Utils";
+import {ICacheContainer} from "../container/ICacheContainer";
+import {CompositeOpBuilder} from "../dmr/CompositeOpBuilder";
+import {IDmrCompositeReq} from "../dmr/IDmrCompositeReq";
+import {JGroupsService} from "../jgroups/JGroupsService";
+import {IServerAddress} from "../server/IServerAddress";
+import {WeakCounter} from "./WeakCounter";
+import {StrongCounter} from "./StrongCounter";
+
+const module: ng.IModule = App.module("managementConsole.services.counter", []);
+
+export class CounterService {
+  static $inject: string[] = ["$q", "jGroupsService", "dmrService", "socketBindingService", "launchType"];
+
+  constructor(private $q: IQService, private jgroupsService: JGroupsService, private dmrService: DmrService,
+              private socketBindingService: SocketBindingService, private launchType: LaunchTypeService) {
+  }
+
+  createStrongCounter(name: string, storage: string, initialValue: number, currentValue: number, lowerBound:number, upperBound: number, dmr: any): ICounter {
+    return new StrongCounter(name, storage, initialValue, currentValue, lowerBound, upperBound, dmr);
+  }
+
+  createWeakCounter(name: string, storage: string, initialValue: number, currentValue: number, concurrency: number, dmr: any): ICounter {
+    return new WeakCounter(name, storage, initialValue, currentValue, concurrency, dmr);
+  }
+
+  doCountersExist(cacheContainer: string, profile: string): ng.IPromise<boolean> {
+    let deferred: ng.IDeferred<boolean> = this.$q.defer<boolean>();
+    let counterAddress: string [] = this.getCountersConfigurationAddress(cacheContainer, profile);
+    let request: IDmrRequest = {
+      address: counterAddress
+    };
+    this.dmrService.readResource(request).then(r => {
+      deferred.resolve(true);
+    }).catch(e => {
+      deferred.resolve(false);
+    });
+    return deferred.promise;
+  }
+
+  createCountersDMRPath(cacheContainer: string, profile: string): ng.IPromise<boolean> {
+    let deferred: ng.IDeferred<boolean> = this.$q.defer<boolean>();
+    let counterAddress: string [] = this.getCountersConfigurationAddress(cacheContainer, profile);
+    let request: IDmrRequest = {
+      address: counterAddress
+    };
+    this.dmrService.add(request).then(r => {
+      deferred.resolve(true);
+    }).catch(e => {
+      deferred.resolve(false);
+    });
```

Comment:
There is no reason for this method to return `IPromise<boolean>` as we never use the return value. Instead we can return IPromise<void> and then have the whole method as:

```typescript
return this.dmrService.add({
      address: this.getCountersConfigurationAddress(cacheContainer, profile)
    });
```